addons/digest/models/digest.py

```diff
@@ -55,20 +56,72 @@ def _compute_available_fields(self):
             digest.available_fields = ', '.join(kpis_values_fields)
 
     def _get_kpi_compute_parameters(self):
-        return fields.Datetime.to_string(self._context.get('start_datetime')), fields.Datetime.to_string(self._context.get('end_datetime')), self.env.company
+        """Get the parameters used to computed the KPI value."""
+        companies = self.company_id
+
+        if any(not digest.company_id for digest in self):
+            # No company: we will use the current company to compute the KPIs
+            companies |= self.env.company
+
+        return (
+            fields.Datetime.to_string(self._context.get('start_datetime')),
+            fields.Datetime.to_string(self._context.get('end_datetime')),
+            companies,
+        )
 
     def _compute_kpi_res_users_connected_value(self):
-        for record in self:
-            start, end, company = record._get_kpi_compute_parameters()
-            user_connected = self.env['res.users'].search_count([('company_id', '=', company.id), ('login_date', '>=', start), ('login_date', '<', end)])
-            record.kpi_res_users_connected_value = user_connected
+        self._compute_company_based_kpi(
+            'res.users',
+            'kpi_res_users_connected_value',
+            date_field='login_date',
+        )
 
     def _compute_kpi_mail_message_total_value(self):
+        start, end, __ = self._get_kpi_compute_parameters()
         discussion_subtype_id = self.env.ref('mail.mt_comment').id
-        for record in self:
-            start, end, company = record._get_kpi_compute_parameters()
-            total_messages = self.env['mail.message'].search_count([('create_date', '>=', start), ('create_date', '<', end), ('subtype_id', '=', discussion_subtype_id), ('message_type', 'in', ['comment', 'email'])])
-            record.kpi_mail_message_total_value = total_messages
+        self.kpi_mail_message_total_value = self.env['mail.message'].search_count([
+            ('create_date', '>=', start),
+            ('create_date', '<', end),
+            ('subtype_id', '=', discussion_subtype_id),
+            ('message_type', 'in', ('comment', 'email')),
+        ])
+
+    def _compute_company_based_kpi(self, model, field_name, date_field='create_date',
```

Comment:
Maybe this could be moved to a "utility" section in the file as it's not actually a compute method in the strict sense? (Or rename but I don't have great suggestions).